File: exception-handler/src/main/java/com/example/exceptionhandler/controller/TestController.java
package com.example.exceptionhandler.controller;

import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class TestController {
    @GetMapping("/exception/{value}")
    public String makeNullPointException(@PathVariable("value") Integer value) {
        if (value == 0) {
            throw new NullPointerException();
        } else if (value == 1) {
            throw new RuntimeException();
        }

        return "Good";
    }

    @ExceptionHandler(NullPointerException.class)
    public Object nullex(Exception e) {
        System.err.println(e.getClass());
        return "myService";
    }
}

File: redis/spring-redis-cache-uid-setting/src/main/kotlin/com/example/springrediscacheuidsetting/controller/TestController.kt
package com.example.springrediscacheuidsetting.controller

import com.example.springrediscacheuidsetting.service.TestService
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RestController

@RestController
class TestController(
    private val testService: TestService,
) {

    @GetMapping("/test/{id}")
    suspend fun call(@PathVariable("id") id: Long) = withContext(Dispatchers.IO) { testService.callValue(id) }

    @GetMapping("/test")
    suspend fun callAll() = withContext(Dispatchers.IO) { testService.callValues() }

    @GetMapping("/mannual/save/{name}")
    suspend fun save(@PathVariable name: String) {
        withContext(Dispatchers.IO) { testService.saveMannual(name) }
    }

    @GetMapping("/mannual/get/{name}")
    suspend fun findVar(@PathVariable name: String): String? {
        return withContext(Dispatchers.IO) {
            testService.findMannual(name)
        }
    }
}